src/driver/kvm.rs

```diff
@@ -474,6 +455,953 @@ mod tests {
         assert!(result.is_ok(), "Expected ok, got error instead!");
     }
 
+    #[test]
+    fn test_fail_to_create_kvm_driver_if_kvmi_init_returns_error() {
+        let mut kvmi_mock = MockKVMi::default();
+        kvmi_mock.expect_init().returning(|_| {
+            Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "something went wrong",
+            ))
+        });
+
+        let result = Kvm::new(
+            "some_vm",
+            kvmi_mock,
+            Some(DriverInitParam::KVMiSocket("/tmp/introspector".to_string())),
+        );
+
+        assert!(result.is_err(), "Expected error, got ok instead!");
+    }
+
+    #[test]
+    fn test_pause_vcpu_when_vm_not_paused() {
+        let mut kvmi_mock = MockKVMi::default();
+        setup_kvmi_mock(&mut kvmi_mock);
+        kvmi_mock.expect_pause().times(1).returning(|| Ok(()));
+        let mut kvm = Kvm::new(
+            "some_vm",
+            kvmi_mock,
+            Some(DriverInitParam::KVMiSocket("/tmp/introspector".to_string())),
+        )
+        .expect("Failed to create driver");
+        let result = Kvm::pause(&mut kvm);
+        assert!(result.is_ok(), "Expected ok, got error instead!");
+    }
+
+    #[test]
+    fn test_pause_vcpu_when_vm_is_paused() {
+        let mut kvmi_mock = MockKVMi::default();
+        setup_kvmi_mock(&mut kvmi_mock);
+        kvmi_mock.expect_pause().times(0).returning(|| Ok(()));
+        let mut kvm = Kvm::new(
+            "some_vm",
+            kvmi_mock,
+            Some(DriverInitParam::KVMiSocket("/tmp/introspector".to_string())),
+        )
+        .expect("Failed to create driver");
+        kvm.expect_pause_ev = 1;
```

Comment:
I would like to avoid manipulating internal variables. Unit tests should try to verify that the public interface of a class behaves as expected. To this end, private functions are usually not tested explicitely and the internal state of an object should only be modified through external influence. I would rename the test to something like `test_pause_vm_only_once_for_multiple_invocations()` and then test accordingly.